wordpress-mobile/WordPress-iOS
# Debug screen for boolean User Defaults

Fixes #22479

This PR introduces a new debug screen for displaying and editing locally stored boolean user defaults. Sometimes it's extremely helpful to change such flags in real-time while testing some features, such as dynamic dashboard cards. The screen has a search feature that filters flags by category/section, key, and site URL.

https://github.com/wordpress-mobile/WordPress-iOS/assets/16563318/197ba38c-303f-4122-9aa9-f7fbb75fbbd1

To test:
- Build a debug app
- Go to Me -> App Settings -> Debug
- [ ] Make sure you can see a new `Boolean User Defaults` section
- Tap on the section and ensure you can see all the stored boolean user defaults
- Smoke test it by editing some flags

## Regression Notes
1. Potential unintended areas of impact
The Debug screen

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual tests

3. What automated tests I added (or what prevented me from doing so)
The screen is for devs only

PR submission checklist:


UI Changes testing checklist:
